server/src/main/java/org/elasticsearch/common/bytes/BytesReference.java

```diff
@@ -184,7 +184,11 @@ static BytesReference fromByteArray(ByteArray byteArray, int length) {
 
     /**
      * Returns a BytesRefIterator for this BytesReference. This method allows
-     * access to the internal pages of this reference without copying them. Use with care!
+     * access to the internal pages of this reference without copying them.
+     * It must return direct references to the pages, not copies. If the object
+     * does not hold internal pages, then it mush return Use with care!
+     *
```

Comment:
Unfinished comment edit I think :)

Reply:
I wanted to add something about implementations without internal pages, I removed it.